torch/csrc/autograd/variable.h

```diff
@@ -110,6 +107,19 @@ struct VariableImpl : public at::TensorImpl {
   friend struct VariableType;
 };
 
+inline Variable make_variable(at::Tensor data) {
+  return Variable(new VariableImpl(data), false);
+}
+
+inline Variable make_variable(at::Tensor data, std::shared_ptr<Function> grad_fn) {
+  return Variable(new VariableImpl(data, grad_fn), false);
```

Comment:
`new VariableImpl(std::move(data), std::move(grad_fn))` (and same with other `make_variable` overloads)

Reply:
I'm going to make these take const references instead. One less move.

Reply:
I don't think that's what you want. That will still invoke the copy constructor. 

For example, this code shouldn't result in extra `retain()` calls:

`make_variable(thing_that_returns_tensor())`

Reply:
OK, I guess then we would also need the move version... I'll make a value constructor then.